lib/class-wp-rest-global-styles-controller-gutenberg.php

```diff
@@ -586,12 +586,11 @@ public function get_item_schema() {
 	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
 	 */
 	public function get_theme_item_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-
 		/*
-		 * Verify if the current user has edit_theme_options capability.
-		 * This capability is required to edit/view/delete templates.
+		 * Verify if the current user has edit_posts capability.
+		 * This capability is required to view global styles.
 		 */
-		if ( ! current_user_can( 'edit_theme_options' ) ) {
+		if ( ! current_user_can( 'edit_posts' ) ) {
```

Comment:
To allow the `canUser( 'read', { kind: 'root', name: 'globalStyles', id: _globalStylesId, } )`useGlobalStylesUserConfig()` check to pass we might need to update the read capabilities of the `wp_global_styles` custom post type:

https://github.com/ramonjd/wordpress-develop/blob/30523ee4b6a0d33b1c2bd8c9ca4021b6256a96c3/src/wp-includes/post.php#L492-L492

E.g., to `'read'                   => 'edit_posts',`

That would mean that editors/authors would be able to read the global styles post, e.g., `await wp.data.resolveSelect( 'core' ).getEntityRecord( 'root', 'globalStyles', await wp.data.resolveSelect( 'core' ).__experimentalGetCurrentGlobalStylesId() )`

It seems harmless enough, but I'm not sure what's kosher.

To display the values, the current editor/author needs to have read access to both.

I haven't thought this through, but I wonder if a new read-only endpoint to retrieve merged theme data might be in order. The `edit-post` package could call it directly from the core store and there'd be no need to shuffle around block settings.

Reply:
Yeah, I think if we think that we should allow "read" for global styles objects using the `edit_posts` capability, we should just make the update in the existing endpoint instead of creating a new one. 

It seems ok to me to do so. I don't see any harm for "editors" to have access to this information (without the ability to modify it)

Reply:
Just commenting on the REST API side of it, that makes sense to me. We do need to adapt the permission check to not just be `edit_posts` though, it needs to include any `show_in_rest` cpt. See `\WP_REST_Themes_Controller::check_read_active_theme_permission` for an example.

Reply:
Thanks for the feedback!

Reply:
Yep, exactly.